src/synapse/azext_synapse/commands.py

```diff
@@ -42,26 +53,38 @@ def load_command_table(self, _):
         g.custom_command('list', 'list_workspaces')
         g.custom_command('create', 'create_workspace', supports_no_wait=True)
         g.custom_command('update', 'update_workspace', supports_no_wait=True)
+        g.custom_command('check-name', 'custom_check_name_availability',
+                         command_type=synapse_operations_sdk,
+                         client_factory=cf_synapse_client_operations_factory)
         g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
         g.wait_command('wait')
 
     with self.command_group('synapse spark pool', synapse_bigdatapool_sdk,
                             client_factory=cf_synapse_client_bigdatapool_factory) as g:
         g.custom_show_command('show', 'get_spark_pool')
-        g.command('list', 'list_by_workspace')
+        g.custom_command('list', 'list_spark_pool_by_workspace')
         g.custom_command('create', 'create_spark_pool', supports_no_wait=True)
         g.custom_command('update', 'update_spark_pool')

```

Comment:
> Should we support no wait for `spark pool update`?

`spark pool update` is not a long running operation. So we shouldn't support no_wait for it.

Reply:
I see. Thanks for pointing it out. It looks like you are using `client.update` rather than `client.create_or_update`. It seems if `client.update` is used then users won't be able to switch auto-scale and auto-pause settings. So I would suggest to switch to `client.create_or_update` and it is a LRO.

Reply:
Now I have switched it to `create_or_update `

---
In reply to: [414297774](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r414297774) [](ancestors = 414297774)